### ISSUE TITLE
FreeBSD 10: Minion not starting properly, hogging CPU

### ISSUE THREAD

**Issue by heini** [saltstack/salt#21180]
Hi,

I've built a current Salt 2014.7.2 from ports on a FreeBSD 10 VM and configured rc.conf to enable the minion. However, "service salt_minion start" prints "Starting salt_minion", but doesn't return and the minion process hogs the CPU at ~99%. I need to explicitely kill the process from another terminal to get the prompt back (CTRL-C doesn't work).

If I run it from the shell, I get this output:

``` bash
# salt-minion -l debug
[DEBUG   ] Reading configuration from /usr/local/etc/salt/minion
[DEBUG   ] Using cached minion ID from /usr/local/etc/salt/minion_id: host.domain
[DEBUG   ] Configuration file path: /usr/local/etc/salt/minion
```

where it hangs forever and brings the CPU to 99% again.

Bye...

Dirk

**Comment by jfindlay** [comment]
@heini, thanks for the report.  What is your minion config?  Can you run `salt-minion -l trace` or `strace salt-minion`?  These may reveal something obviously ridiculous being done by salt.

**Comment by heini** [comment]
My minion config is an unmodified copy of the sample config.

```bash
# salt-minion -l trace
[DEBUG   ] Reading configuration from /usr/local/etc/salt/minion
[DEBUG   ] Using cached minion ID from /usr/local/etc/salt/minion_id: freebsd.altum.de
[TRACE   ] loading log_handlers in ['/var/cache/salt/minion/extmods/log_handlers', '/usr/local/lib/python2.7/site-packages/salt/log/handlers']
[TRACE   ] Skipping /var/cache/salt/minion/extmods/log_handlers, it is not a directory
[TRACE   ] None of the required configuration sections, 'logstash_udp_handler' and 'logstash_zmq_handler', were found the in the configuration. Not loading the Logstash logging handlers module.
[DEBUG   ] Configuration file path: /usr/local/etc/salt/minion
[TRACE   ] Trying pysss.getgrouplist for 'root'
```

This already rings a bell (the pysss bit). I'm using sssd to authenticate against ldap/kerberos5. However, this shouldn't affect root...

Can't install strace, though:
```bash
# make install clean                                                                                                  :(
===>  strace-4.5.18_1 is only for i386, while you are running amd64.
*** Error code 1

Stop.
make: stopped in /usr/ports/devel/strace
```

**Comment by jfindlay** [comment]
@heini, thanks for the extra info.  Is that last statement in the trace log the final log statement before the minion maxes out the processor?

**Comment by heini** [comment]
Yes.

**Comment by cachedout** [comment]
Could you be hitting this? http://www.itsecdb.com/oval/definition/oval/org.mitre.oval/def/26898/ELSA-2013-1319----sssd-security-and-bug-fix-update-low-.html

**Comment by heini** [comment]
Don't think so. sssd version is 1.11.7, which is far newer than 1.5.1.

**Comment by castillar** [comment]
Hi all,

I just got hit by this bug, and I'm happy to see I'm not alone! My system is a FreeBSD 10.3-RELEASE-p7 node, and I had sssd installed at the same version @heini reported (1.11.7_4). My salt-minion process would also start up and halt, burning 100% of CPU; running with debug enabled produced the same last message about pysss.getgrouplist.

I had sssd installed for a previous experiment and didn't need it so I decided to try uninstalling it, which worked. Specifically, I did this:

1. Reconfigure/reinstall sudo to disable the "Enable SSSD backend support" flag.
2. Remove sssd ("pkg remove sssd") from the system.
3. Reinstall py-salt (presumably this removed the SSSD libraries in the process--I tried restarting salt-minion before doing this, but it didn't work without doing this).
4. Restart salt-minion.

Doing this worked cleanly and resulted in a functional salt-minion on this host. Looks like maybe some conflict between Salt and the SSSD libraries?
